Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -6971,13 +6947,26 @@ void WebPageProxy::decidePolicyForNavigationActionSync(NavigationActionPolicyPar
 
 void WebPageProxy::decidePolicyForNavigationActionSyncShared(Ref<WebProcessProxy>&& process, PageIdentifier webPageID, NavigationActionPolicyParameters&& parameters, CompletionHandler<void(PolicyDecision&&)>&& reply)
 {
-    auto sender = PolicyDecisionSender::create(WTFMove(reply));
-
     RefPtr frame = WebFrameProxy::webFrame(parameters.frameInfo.frameID);
     MESSAGE_CHECK(process, frame);
 
+    struct PolicyDecisionSender : public RefCounted<PolicyDecisionSender> {
```

Comment:
I think this should be a class with a public `create()` function and a private constructor. Refcounted classes with a public constructor are too error-prone, which is why we don't use them.